example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignFlow.kt

```diff
@@ -0,0 +1,189 @@
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.MediaType.APPLICATION_PDF_VALUE
+import org.springframework.http.MediaType.TEXT_HTML
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.BodyInserters.fromFormData
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.reactive.function.client.awaitBody
+import org.springframework.web.server.ServerWebExchange
+import java.net.URI
+
+@RestController
+class TruidSignFlow(
+    @Value("\${oauth2.clientId}")
+    val clientId: String,
+
+    @Value("\${oauth2.clientSecret}")
+    val clientSecret: String,
+
+    @Value("\${oauth2.truid.sign-endpoint}")
+    val truidSignupEndpoint: String,
+
+    @Value("\${oauth2.truid.token-endpoint}")
+    val truidTokenEndpoint: String,
+
+    @Value("\${oauth2.truid.signature-endpoint}")
+    val truidSignatureEndpoint: String,
+
+    @Value("\${app.domain}")
+    val publicDomain: String,
+
+    @Value("\${web.sign.success}")
+    val webSuccess: URI,
+
+    @Value("\${web.sign.failure}")
+    val webFailure: URI,
+
+    val webClient: WebClient
+) {
+    @GetMapping("/documents/Agreement.pdf", produces = [APPLICATION_PDF_VALUE])
+    suspend fun document(): ByteArray {
+        return getDocument()
+    }
+
+    @GetMapping("/truid/v1/sign")
+    suspend fun sign(
+        @RequestHeader("X-Requested-With") xRequestedWith: String?,
+        exchange: ServerWebExchange
+    ) {
+        val session = exchange.session.awaitSingle()
+        val document = getDocument()
+
+        val truidSignupUrl = URIBuilder(truidSignupEndpoint)
+            .addParameter("response_type", "code")
+            .addParameter("client_id", clientId)
+            .addParameter("scope", "truid.app/data-point/email")
+            .addParameter("redirect_uri", "$publicDomain/truid/v1/complete-sign")
+            .addParameter("state", createOauth2State(session))
+            .addParameter("code_challenge", createOauth2CodeChallenge(session))
+            .addParameter("code_challenge_method", "S256")
+            .addParameter("user_message", "Please sign this document")
+            .addParameter("data_object_id", "$publicDomain/documents/Agreement.pdf")
+            .addParameter("data_object_digest", base64url(sha256(document)))
+            .addParameter("data_object_digest_algorithm", "S256")
+            .addParameter("data_object_b64", "false")
+            .addParameter("data_object_content_type", "application/pdf")
```

Comment:
The etsi/jades recommends to skip the `application/` part. We (truid) only passes it on, so it doesn't matter for us.
Not sure what we should recommend to our customers.. .
I agree it makes more sense to include the full content type..

Reply:
I think that we should check if the content type starts with `application/` and strip that part in that case.

Because the standard that we refer to here sais "use application/pdf", it sais nothing about stripping anything.
We are taking this value and putting it in the other standard that sais "strip application/". So we should strip it.